middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -697,27 +752,42 @@ let close_c_call acc env ~loc ~let_bound_ids_with_kinds
         (Acc.with_free_names free_names_of_body acc),
       body )
   in
-  let box_unboxed_returns ~let_bound_var ~box_return_value =
-    let let_bound_var' = VB.create let_bound_var Name_mode.normal in
-    let handler_param = Variable.rename let_bound_var in
+  let box_unboxed_returns () =
+    let let_bound_vars' =
+      List.map
+        (fun let_bound_var -> VB.create let_bound_var Name_mode.normal)
+        let_bound_vars
+    in
+    let handler_params =
+      List.map
+        (fun let_bound_var -> Variable.rename let_bound_var)
+        let_bound_vars
+    in
     let body acc =
       let acc, body = keep_body acc in
-      let named =
-        Named.create_prim
-          (Unary (box_return_value, Simple.var handler_param))
-          dbg
-      in
-      Let_with_acc.create acc
-        (Bound_pattern.singleton let_bound_var')
-        named ~body
+      List.fold_left2
+        (fun (acc, body) unarized_param (handler_param, let_bound_var') ->
+          match unarized_param.return_transformer with
+          | None -> acc, body
```

Comment:
I think we need to bind `let_bound_var'` to `handler_param` here, or ensure we don't perform a rename of this `handler_param` in that case.

Reply:
Well spotted, fixed.  I think we're going to add tests in the subsequent PR but will confirm.